File: python/collatz-conjecture/collatz_conjecture.py
def isEven(num):
    return num % 2 == 0


def steps(number):
    if number < 1:
        raise ValueError("Only positive integers are allowed")

    count = 0
    while number > 1:
        if isEven(number):
            number //= 2
        else:
            number = 3 * number + 1
        count += 1
    return count

File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        int count = 0;
        int num = number;

        if (num < 1)
        {
            throw new ArgumentOutOfRangeException();
        }

        while (num > 1)
        {
            if (num % 2 == 0)
            {
                num = num / 2;
            }
            else
            {
                num = 3 * num + 1;
            }
            ++count;
        }
        return count;
    }
}
